fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -2994,6 +2995,38 @@ public void alterMaterializedView(AlterMaterializedViewStatement stmt) throws Dd
         stateMgr.getAlterInstance().processAlterMaterializedView(stmt);
     }
 
+    @Override
+    public void refreshMaterializedView(RefreshMaterializedViewStatement stmt)
+            throws DdlException, MetaNotFoundException {
+        final String mvName = stmt.getMvName().getTbl();
+        final String dbName = stmt.getMvName().getDb();
+        Database db = this.getDb(dbName);
+        if (db == null) {
+            ErrorReport.reportDdlException(ErrorCode.ERR_BAD_DB_ERROR, dbName);
+        }
+        MaterializedView materializedView = null;
+        db.readLock();
+        try {
+            final Table table = db.getTable(mvName);
+            if (table instanceof MaterializedView) {
+                materializedView = (MaterializedView) table;
+            }
+        } finally {
+            db.readUnlock();
+        }
+        if (materializedView == null) {
+            throw new MetaNotFoundException("Materialized view " + mvName + " is not found");
+        }
+        TaskManager taskManager = GlobalStateMgr.getCurrentState().getTaskManager();
+        final String mvTaskName = TaskBuilder.getMvTaskName(materializedView.getId());
+        if (!taskManager.getNameToTaskMap().containsKey(mvTaskName)) {
```

Comment:
better to add taskManager containsTask method，and not expose data structures NameToTaskMap